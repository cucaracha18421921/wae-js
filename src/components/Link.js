import React from 'react'
import PropTypes from 'prop-types'
import {Button} from "./uiComponents";
import styled from "styled-components";

const LinkButton = styled(Button)`
  margin-left: 4px;
  &:disabled {
    background-color: rgba(0,0,0,0.1);
  }
`;

const Link = ({ active, children, onClick }) => (
    <LinkButton
       onClick={onClick}
       disabled={active}
    >
      {children}
    </LinkButton>
)

Link.propTypes = {
  active: PropTypes.bool.isRequired,
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func.isRequired
}

export default Link
