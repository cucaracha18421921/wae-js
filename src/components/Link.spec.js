import {shallow} from "enzyme";
import Link from "./Link";
import React from "react";

let wrapper;

beforeEach(() => {
  wrapper = shallow(<Link active={true} onClick={()=>{}} children={[]} />);
});

describe('Link rendering', () => {
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
