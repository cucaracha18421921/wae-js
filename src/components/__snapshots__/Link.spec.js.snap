// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link rendering renders correctly 1`] = `ShallowWrapper {}`;
